pkg/utils/kube/ctl.go

```diff
@@ -38,7 +38,6 @@ type OnKubectlRunFunc func(command string) (CleanupFunc, error)
 
 type Kubectl interface {
 	ApplyResource(ctx context.Context, config *rest.Config, obj *unstructured.Unstructured, namespace string, dryRunStrategy cmdutil.DryRunStrategy, force, validate bool) (string, error)
-	ConvertToVersion(obj *unstructured.Unstructured, group, version string) (*unstructured.Unstructured, error)
```

Comment:
vX->vY conversion only works on known objects, does not work on CRDs. Also, no need to do it here since the engine is watching all resources anyway - is that right? Or what are the cases when conversion is needed?